powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/persistence/OperationQueryService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2024 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.app.server.service.persistence;
+
+import io.getlime.security.powerauth.app.server.database.model.entity.OperationEntity;
+import org.springframework.data.domain.Pageable;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * Service for operation queries with pessimistic locking.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public interface OperationQueryService {
+
+    Optional<OperationEntity> findOperationForUpdate(String operationId);
+
+    Optional<OperationEntity> findOperationWithoutLock(String operationId);
```

Comment:
May we have a JavaDoc at least for the first two methods?

Reply:
Yes, I will write Javadoc for all methods.